core/controllers/reader_test.py

```diff
@@ -362,3 +363,169 @@ def test_ratings_by_different_users(self):
             ratings['overall_ratings'],
             {'1': 0, '2': 0, '3': 0, '4': 2, '5': 0})
         self.logout()
+
+
+class FlagExplorationHandlerTests(test_utils.GenericTestBase):
+
+    EXP_ID = '0'
+
+    def setUp(self):
+        super(FlagExplorationHandlerTests, self).setUp()
+
+        # Register users.
+        self.signup(self.MODERATOR_EMAIL, self.MODERATOR_USERNAME)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.new_user_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+        self.moderator_id = self.get_user_id_from_email(self.MODERATOR_EMAIL)
+        self.set_moderators([self.MODERATOR_USERNAME])
+
+        # Load exploration 0.
+        exp_services.delete_demo(self.EXP_ID)
```

Comment:
I think this line isn't needed. Feel free to (optionally) remove it elsewhere if it's similarly superfluous there.